processor/k8sprocessor/processor.go

```diff
@@ -122,119 +70,83 @@ func (kp *kubernetesprocessor) Shutdown(context.Context) error {
 	return nil
 }
 
-func (kp *kubernetesprocessor) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
+// ProcessTraces process traces and add k8s metadata using resource IP or incoming IP as pod origin.
+func (kp *kubernetesprocessor) ProcessTraces(ctx context.Context, td pdata.Traces) (pdata.Traces, error) {
 	rss := td.ResourceSpans()
 	for i := 0; i < rss.Len(); i++ {
 		rs := rss.At(i)
 		if rs.IsNil() {
 			continue
 		}
 
-		var podIP string
-		resource := rs.Resource()
-
-		// check if the application, a collector/agent or a prior processor has already
-		// annotated the batch with IP.
-		if !resource.IsNil() {
-			podIP = kp.k8sIPFromAttributes(resource.Attributes())
-		}
-
-		// Check if the receiver detected client IP.
-		if podIP == "" {
-			if c, ok := client.FromContext(ctx); ok {
-				podIP = c.IP
-			}
-		}
-
-		if podIP != "" {
-			if resource.IsNil() {
-				resource.InitEmpty()
-			}
-			resource.Attributes().InsertString(k8sIPLabelName, podIP)
-		}
+		_ = kp.processResource(ctx, rs.Resource(), k8sIPFromAttributes())
```

Comment:
Should catch and log the error at least.

Reply:
Since `processResource` never returns anything other than `nil`, I removed `error` from return type